internal/states/statefile/version4.go

```diff
@@ -710,7 +740,7 @@ type resourceStateV4 struct {
 	Type           string                  `json:"type"`
 	Name           string                  `json:"name"`
 	EachMode       string                  `json:"each,omitempty"`
-	ProviderConfig string                  `json:"provider"`
+	ProviderConfig string                  `json:"provider,omitempty"` // Ronny TODO: Do we want to omit? Or save an empty string / nil?
```

Comment:
This will affect whether we see the `provider` attribute on the resource in the state file

Reply:
IMO we have to ensure other tools **fail** correctly if this field is something they rely on. I think it should be omit completely.